pkg/types/validation/installconfig.go

```diff
@@ -571,3 +578,27 @@ func validateURI(uri string, fldPath *field.Path, schemes []string) field.ErrorL
 	}
 	return field.ErrorList{field.NotSupported(fldPath, parsed.Scheme, schemes)}
 }
+
+// validateIPProxy checks if the given proxy string is an IP and if so checks the service and
+// cluster networks and returns error if the IP belongs in them. Returns nil if the proxy is
+// not an IP address.
+func validateIPProxy(proxy string, n *types.Networking, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	proxyIP := net.ParseIP(proxy)
+	if proxyIP == nil {
+		return nil
+	}
+	for _, network := range n.ClusterNetwork {
+		if network.CIDR.Contains(proxyIP) {
+			allErrs = append(allErrs, field.Invalid(fldPath, proxy, "proxy value is part of the cluster networks"))
```

Comment:
If the CIDRs overlap, then the proxy IP could be contained in multiple CIDRs. This would result in the same error duplicated. The CIDRs are not allowed to overlap and should be caught by other validation, but this validation runs at the same time.